ui/apps/ui/src/app/components/sort-by-functionality/sort-by-functionality.component.ts

```diff
@@ -0,0 +1,105 @@
+import { Component, OnInit } from '@angular/core';
+import { FormControl } from '@angular/forms';
+import { Router } from '@angular/router';
+import { UntilDestroy, untilDestroyed } from '@ngneat/until-destroy';
+import { CustomRoute } from '@collections/services/custom-route.service';
+import { sanitizeQuery } from '@components/search-input/query.sanitizer';
+import { filter, take } from 'rxjs';
+import { DEFAULT_SORT } from '@components/sort-by-functionality/sort-value-default';
+
+@UntilDestroy()
+@Component({
+  selector: 'ess-sort-by-functionality',
+  template: `<div class="col-sm-4 sort_container">
+    <label for="sorts" i18n>Sort By</label>
+    <select
+      [formControl]="selectedSortOptionControl"
+      id="sorts"
+      class="form-select"
+    >
+      <option value="r" i18n>Relevancy</option>
+      <option value="dmr" i18n>Date - Most recent</option>
+      <option value="dlr" i18n>Date – Least recent</option>
+      <option value="mp" i18n>Most popular</option>
+    </select>
+  </div>`,
+  styles: [
+    `
+      :host {
+        display: contents;
+      }
+
+      label {
+        width: 100px;
+        font-family: 'Inter';
+        font-style: normal;
+        font-weight: 400;
+        font-size: 14px;
+        line-height: 20px;
+        color: #4f4f4f;
+      }
+
+      select {
+        font-family: 'Inter';
+        font-style: normal;
+        font-weight: 500;
+        font-size: 14px;
+        line-height: 24px;
+        color: #101828;
+      }
+
+      select:focus {
+        -webkit-box-shadow: none;
+        box-shadow: none;
+        border: 1px solid #ced4da;
+      }
+
+      .sort_container {
+        width: 18rem;
+        display: flex;
+        align-items: center;
+      }
+    `,
+  ],
+})
+export class SortByFunctionalityComponent implements OnInit {
+  public selectedSortOptionControl: FormControl = new FormControl(DEFAULT_SORT);
+
+  constructor(private _customRoute: CustomRoute, private _router: Router) {}
+
+  ngOnInit() {
+    this._customRoute.s$
+      .pipe(
+        untilDestroyed(this), // TODO: event triggered on clear or change search criteria
+        filter((s) => s !== this.selectedSortOptionControl.value)
+      )
+      .subscribe((s) => {
+        this.selectedSortOptionControl.setValue(s, { emitEvent: false });
+      });
```

Comment:
Here some typing could be used if we have form control which is strongly typed:

```typescript
    this._customRoute.s$
      .pipe(
        untilDestroyed(this), // TODO: event triggered on clear or change search criteria
        filter((s) => s !== this.selectedSortOptionControl.value)
      )
      .subscribe((s) => {
        this.selectedSortOptionControl.setValue(
          isSortOption(s) ? s : DEFAULT_SORT,
          { emitEvent: false }
        );
      });

```

The `isSortOption` is guard option which could be used to make sure that `s` is a given subtype. (It could also be placed in `sort-value-default.ts`)

```typescript
export const DEFAULT_SORT = 'dlr';
export type sortType = 'r' | 'dmr' | 'dlr' | 'mp';


export function isSortOption(obj: string): obj is sortType {
  return ['dlr', 'dmr', 'mp', 'r'].includes(obj);
}

```

Reply:
Nice. done.